### ISSUE TITLE
Handle absent/broken audio device on startup

### ISSUE THREAD

**Issue by badosu** [LMMS/lmms#1847]
Previously, we relied on showing the user the actual device being used on the settings window, however this was not a good idea as it had ambiguity with the settings management, which was addressed on this PR: https://github.com/LMMS/lmms/pull/1832.

We should find a way to detect on LMMS load that no audio device is connected and alert the user instead.

**Comment by tresf** [comment]
IIRC, the settings dialog opens in this use-case.

Are you saying that won't happen now?

**Comment by badosu** [comment]
No, I am saying that the settings dialog will show the device written on the file and not alerting the user if lmms couldn't use the device/backend.

What I am suggesting is that we should alert the user when we have trouble configuring the audio backend. Previously the Dummy output was always selected when this happened and the user could figure out when something wrong happened, but after #1832 he does not have this indication available anymore.

**Comment by tresf** [comment]
Ah yes.  :)

**Comment by badosu** [comment]
This should be really easy for a beginner to solve, just replace `printf` with `Song::collectError` on https://github.com/LMMS/lmms/blob/master/src/core/Mixer.cpp#L830-L832

See example usage at https://github.com/LMMS/lmms/blob/master/plugins/LadspaEffect/LadspaEffect.cpp#L78-79

**Comment by softrabbit** [comment]
@badosu, not quite that easy. Collecting the error is only half the solution, the error needs to be shown to the user to be of any use.

**Comment by badosu** [comment]
@softrabbit Please see: https://github.com/LMMS/lmms/blob/master/src/core/Song.cpp#L1053-L1064

If the message is collected at song initialization it will be shown.

**Comment by softrabbit** [comment]
Well, the errors are cleared at the start of loading the song: https://github.com/LMMS/lmms/blob/master/src/core/Song.cpp#L952 so anything collected before that is lost. Connecting to the audio device happens earlier. 

Anyway, I feel falling back to dummy output and silent operation is serious enough to show a message box of its own, maybe using ```QMessageBox::Critical``` for an icon.

**Comment by badosu** [comment]
Indeed, well observed.

Well, so we have 2 options:

1. Decouple the UI notification logic from the Song
2. Create a specific logic for this case

I'd prefer (1)

**Comment by softrabbit** [comment]
With the feature freeze coming up, I think some kind of immediate notification, AKA option 2, is the way to go, leaving the almighty log and notification infrastructure to be created for 1.3 or 2.0... or whatever it will be called.

And BTW, I just got sound out even if both my settings dialog and lmmsrc.xml say "dummy"...
